lib/ocpp/v201/charge_point.cpp

```diff
@@ -3194,6 +3209,178 @@ void ChargePoint::handle_customer_information_req(Call<CustomerInformationReques
     }
 }
 
+void ChargePoint::handle_set_monitoring_base_req(Call<SetMonitoringBaseRequest> call) {
+    SetMonitoringBaseResponse response;
+    const auto& msg = call.msg;
+
+    bool valid_monitor = false;
+    switch (msg.monitoringBase) {
+    case MonitoringBaseEnum::All:
+    case MonitoringBaseEnum::FactoryDefault:
+    case MonitoringBaseEnum::HardWiredOnly:
+        valid_monitor = true;
+        break;
+    }
+
+    if (valid_monitor) {
+        auto result = this->device_model->set_value(
+            ControllerComponentVariables::ActiveMonitoringBase.component,
+            ControllerComponentVariables::ActiveMonitoringBase.variable.value(), AttributeEnum::Actual,
+            conversions::monitoring_base_enum_to_string(msg.monitoringBase), true);
+
+        if (result != SetVariableStatusEnum::Accepted) {
+            EVLOG_warning << "Could not persist in device model new monitoring base: "
+                          << conversions::monitoring_base_enum_to_string(msg.monitoringBase);
+            response.status = GenericDeviceModelStatusEnum::Rejected;
+        } else {
+            response.status = GenericDeviceModelStatusEnum::Accepted;
+
+            // TODO(ioan): Spec 3.56: see if we need to clear custom monitors (the ones
+            // set by the CSMS) on 'FactoryDefault' too
+            if (msg.monitoringBase == MonitoringBaseEnum::HardWiredOnly) {
+                try {
+                    this->device_model->clear_custom_monitors();
+                } catch (const QueryExecutionException& e) {
+                    EVLOG_warning << "Could not clear custom monitors from DB: " << e.what();
+                    response.status = GenericDeviceModelStatusEnum::Rejected;
+                }
```

Comment:
The device model storage implementation currently doesn't throw `QueryExecutionException`, this is to be added in a subsequent PR 

We have an abstraction layer for the DeviceModelStorage, so we probably do not want to define Device Model specific exceptions for these cases